### PR TITLE
9.51.4

### PR DESCRIPTION
**Pull request by robby-dermody**
@adamkrellenstein @ouziel-slama 

this code has been running well on cw01 for about 5 or 6 days now. no crashes, and the availability issues I was seeing have stopped

undolog support is not added into this (that will be in next release), so this is the performance code that ouziel has already reviewed, plus some additional enhancements. I don't believe it touches any consensus code, but we should all obviously double-check

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -1,4 +1,12 @@
 ## Client Versions ##
+* v9.51.4 (2015-09-22)
+    * Significant performance and caching optimizations around bitcoind addrindex interaction
+    * Fixed issues around responsiveness issues with larger mempool sizes
+    * Mempool refresh period changed from 2 seconds down to .5 seconds
+    * Bigfixes with coinbase transaction processing
```
Comment (adamkrellenstein) [inline ChangeLog.md]: 'bigfixes' -> 'big fixes'

File: ChangeLog.md [hunk 2]
```diff
@@ -1,4 +1,12 @@
 ## Client Versions ##
+* v9.51.4 (2015-09-22)
+    * Significant performance and caching optimizations around bitcoind addrindex interaction
+    * Fixed issues around responsiveness issues with larger mempool sizes
+    * Mempool refresh period changed from 2 seconds down to .5 seconds
```
Comment (adamkrellenstein) [inline ChangeLog.md]: 'down to' -> 'to'

File: ChangeLog.md [hunk 3]
```diff
@@ -1,4 +1,12 @@
 ## Client Versions ##
+* v9.51.4 (2015-09-22)
+    * Significant performance and caching optimizations around bitcoind addrindex interaction
+    * Fixed issues around responsiveness issues with larger mempool sizes
```
Comment (adamkrellenstein) [inline ChangeLog.md]: 'responsiveness issues' -> 'responsiveness'

**Comment by adamkrellenstein** [comment]
ACK

**Comment by adamkrellenstein** [comment]
@ouziel-slama should have a second look, as well

**Comment by ouziel-slama** [comment]
ACK

**Comment by robby-dermody** [comment]
@adamkrellenstein good to merge?

**Comment by adamkrellenstein** [comment]
Yep. Lemme do that now.

**Pull request merged**